packages/upstream-protobuf/bin/protoc-gen-dumpcodegenreq.mjs

```diff
@@ -21,8 +21,21 @@ import { stdin, stdout } from "node:process";
 stdin.pipe(createWriteStream("dumpcodegenreq.binpb"));
 
 // Write a minimal a google.protobuf.CodeGenerateResponse to stdout.
-// This is just the field supported_features set to a bitwise "or" of
-// FEATURE_PROTO3_OPTIONAL and FEATURE_SUPPORTS_EDITIONS. See the code snippet
-// below.
-const minimalResponse = new Uint8Array([16, 3]);
+/*
+import {create, toBinary} from "@bufbuild/protobuf";
+import {CodeGeneratorResponse_Feature, CodeGeneratorResponseDesc, Edition} from "@bufbuild/protobuf/wkt";
+const res = create(CodeGeneratorResponseDesc, {
+  supportedFeatures: BigInt(
+    CodeGeneratorResponse_Feature.SUPPORTS_EDITIONS | CodeGeneratorResponse_Feature.PROTO3_OPTIONAL
+  ),
+  minimumEdition: Edition.EDITION_LEGACY,
+  maximumEdition: Edition.EDITION_MAX,
+});
+console.log(
+  toBinary(CodeGeneratorResponseDesc, res)
+);
+*/
+const minimalResponse = new Uint8Array([
+  16, 3, 24, 132, 7, 32, 255, 255, 255, 255, 7,
+]);
```

Comment:
This script is used internally to simulate plugin invocations. We simply dump the code generator request.

`protoc` v27.0-rc1 starts honoring the fields `minimum_edition` and  `maximum_edition` of google.protobuf.compiler.CodeGeneratorResponse, and errors if a file uses an edition that the plugin does not advertise as supported.

We don't want to use our runtime here to create a response, because we use this code to test our runtime. So we manually create a response that advertises support for all editions, using the range Edition.EDITION_LEGACY to Edition.EDITION_MAX, and use the binary data literally here.